### PR TITLE
Backup: Update component to use custom backup retention days if set

### PR DESCRIPTION
**Pull request by rustamshaikh**
## Proposed changes:

Update the component to make use of a custom backup retention period (in days) if set by the customer otherwise use the retention from the plan.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1HpG7-jZR-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Take your own site whose backups are being discarded due to the storage limit having been reached and does not have a custom retention period set yet.
- Now set a custom retention period, specifically lower the retention period say from 30 days to 7 days via Jetpack Cloud or `vpshell`.
- Check that both the Jetpack cloud and Jetpack plugin show the same backup usage status i.e. under storage limit and normal. 

Similar to https://github.com/Automattic/wp-calypso/pull/73007

### CODE REVIEW

File: projects/plugins/videopress/jetpack-videopress.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * Plugin Name: Jetpack VideoPress
  * Plugin URI: https://wordpress.org/plugins/jetpack-videopress
  * Description: High quality, ad-free video.
- * Version: 1.4.0
```
Comment (Initsogar) [inline projects/plugins/videopress/jetpack-videopress.php]: I'm not sure whether we should update the VideoPress version for these changes. I think that using `tools/project-version.sh -f -u 1.12.4-alpha packages/backup` should do the trick to bump only the backup package.

**Review by Initsogar — DISMISSED** [review]
Looks good to me! Tested and works as expected.
I just left a comment about the VideoPress plugin version bump that I don't think is needed for these changes.

Adding `Needs review` label so Jetpack Crew could take a look before merging.

File: projects/plugins/videopress/jetpack-videopress.php (hunk 1 above)
Comment (rustamshaikh) [reply projects/plugins/videopress/jetpack-videopress.php]: Sometime before, I ran `tools/fixup-project-versions.sh` to fix the versions but unknowingly it also touched Videopress files which later forced me to push a changelog for Videopress as well. Will take a look again if I could revert those.

Comment (Initsogar) [reply projects/plugins/videopress/jetpack-videopress.php]: Yeah, I will suggest cleaning it up, rebase/merge from trunk and then try again maybe `tools/project-version.sh -f -u 1.12.4-alpha packages/backup`.

I think the changelogger checks for missing version, considering the trunk as well.

Comment (rustamshaikh) [reply projects/plugins/videopress/jetpack-videopress.php]: Done ✅

**Review by Initsogar — DISMISSED** [review]
Thanks for fixing the additional changelog files! Approved.

**Review by Initsogar — APPROVED** [review]
Updated package version to `1.12.5-alpha` due to a conflict with trunk.

**Pull request merged**